ingest/expression_files/mtx.py

```diff
@@ -183,64 +194,96 @@ def transform(self):
+                            yield gene_models, GeneExpression.COLLECTION_NAME
+                            yield data_arrays, DataArray.COLLECTION_NAME
+                            num_processed += len(gene_models)
+                            print(
+                                f"Processed {num_processed} genes. "
+                                f"{str(datetime.datetime.now() - start_time)} "
+                                f"elapsed"
+                            )
+                            gene_models = []
+                            data_arrays = []
+                        data_arrays += current_data_arrays
                         # Reset variables so values will be associated w/new
                         # gene
                         exp_cells = []
                         exp_scores = []
-                if len(data_arrays) > 1_000:
-                    yield gene_models, data_arrays
-                    num_processed += len(gene_models)
-                    print(
-                        f"Processed {num_processed} genes. "
-                        f"{str(datetime.datetime.now() - start_time)} "
-                        f"elapsed"
+                    model_id = ObjectId()
+                    gene_model = GeneExpression.create_gene_model(
+                        name=gene,
+                        study_file_id=self.study_file_id,
+                        study_id=self.study_id,
+                        gene_id=gene_id,
+                        _id=model_id,
                     )
-                    num_processed += len(gene_models)
-                    gene_models = []
-                    data_arrays = []
-                model_id = ObjectId()
-                # Add current gene's gene model
-                gene_model = GeneExpression.create_gene_model(
-                    name=gene,
-                    study_file_id=self.study_file_id,
-                    study_id=self.study_id,
-                    gene_id=gene_id,
-                    _id=model_id,
-                )
-                gene_models.append(gene_model)
-                last_idx = current_idx
+                    gene_models.append(gene_model)
+                    last_idx = current_idx
             exp_cell = self.cells[int(raw_barcode_idx) - 1]
             exp_score = round(float(raw_exp_score), 3)
             exp_cells.append(exp_cell)
             exp_scores.append(exp_score)
+        # Create data array for last row
+        for data_array in GeneExpression.create_data_arrays(
+            name=f"{gene} Cells",
+            array_type="cells",
+            values=exp_cells,
+            linear_data_type="Gene",
+            linear_data_id=model_id,
+            **self.data_array_kwargs,
+        ):
+            data_arrays.append(data_array)
+        # Data array for expression values
+        for data_array in GeneExpression.create_data_arrays(
+            name=f"{gene} Expression",
+            array_type=f"expression",
```

Comment:
No concatenation, so no f-string:

```suggestion
            array_type="expression",
```